stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -319,42 +375,94 @@ private void runInternal() throws Exception {
-            }
-            long ioBytes = applyOperation(filePath);
-
-            // Done reading/writing one file
-            if (ioBytes <= 0) {
-              break;
-            }
-            // Start recording after the warmup
-            if (CommonUtils.getCurrentMs() > recordMs) {
-              mThreadCountResult.incrementIOBytes(ioBytes);
-            }
+    private void writeOrLocalRead() throws Exception {
+      for (int testDirId = mThreadId; testDirId < mParameters.mNumDirs;
+          testDirId += mParameters.mThreads) {
+        for (int testFileId = 0; testFileId < mParameters.mNumFilesPerDir; testFileId++) {
+          String filePath = String.format(TEST_DIR_STRING_FORMAT + "/file-%d",
+              mParameters.mLocalPath, mBaseParameters.mId, testDirId, testFileId);
+          processFile(filePath, FuseIOOperation.isRead(mParameters.mOperation));
+        }
+      }
+      finishProcessingFiles();
+    }
+
+    private void remoteOrClusterRead() throws Exception {
+      for (int numJobWorkerDirProcessed = 0; numJobWorkerDirProcessed < mJobWorkerDirNames.size();
+          numJobWorkerDirProcessed++) {
+        // find which job worker directory to read
+        int indexCurrentJobWorkerDir = (numJobWorkerDirProcessed + mJobWorkerZeroBasedId)
+            % mJobWorkerDirNames.size();
+        // skip itself if the operation is remote read
+        if (indexCurrentJobWorkerDir == mJobWorkerZeroBasedId
+            && mParameters.mOperation == FuseIOOperation.REMOTE_READ) {
+          indexCurrentJobWorkerDir = (indexCurrentJobWorkerDir + 1) % mJobWorkerDirNames.size();
+        }
+        String nameCurrentJobWorkerDir = mJobWorkerDirNames.get(indexCurrentJobWorkerDir);
+
+        // find which files to read under this job worker directory
+        for (int testDirId = mJobWorkerZeroBasedId; testDirId < mParameters.mNumDirs;
+            testDirId += mJobWorkerDirNames.size()) {
+          for (int testFileId = mThreadId; testFileId < mParameters.mNumDirs;
+              testFileId += mParameters.mThreads) {
+            String filePath = String.format(TEST_DIR_STRING_FORMAT + "/file-%d",
+                mParameters.mLocalPath, mBaseParameters.mId, testDirId, testFileId);
+            processFile(filePath, FuseIOOperation.isRead(mParameters.mOperation));
           }
         }
       }
-      // Done reading all files
-      if (isRead) {
-        throw new IllegalArgumentException(String.format("Thread %d finishes reading all its files "
-            + "before the bench ends. For more accurate result, use more files, or larger files, "
-            + "or a shorter duration", mThreadId));
+      finishProcessingFiles();
+    }
+
+    private void processFile(String filePath, boolean isRead) throws IOException {
+      mCurrentOffset = 0;
+      while (!Thread.currentThread().isInterrupted()) {
+        if (isRead && CommonUtils.getCurrentMs() > mContext.getEndMs()) {
+          closeInStream();
+          return;
```

Comment:
we stop processing this file but didn't stop processing the following files
if current > end, we should stop processing all the files

Reply:
Need a flag to tell the caller whether the return is because of time up or done processing file. If the caller sees time up flag is on, it should stop and also not error out.